File: Loenn/entities/BreakBlock.lua
local fakeTilesHelper = require("helpers.fake_tiles")

local BreakBlock = {}

BreakBlock.name = "XaphanHelper/BreakBlock"
BreakBlock.depth = -13000
BreakBlock.fieldOrder = {
    "x", "y", "width", "height", "flag", "mode", "tiletype", "flagTiletype", "directory", "type", "color", "respawn", "destroyStaticMovers", "startRevealed"
}
function BreakBlock.fieldInformation(entity)
    return {
        tiletype = {
            options = fakeTilesHelper.getTilesOptions(),
            editable = false
        },
        flagTiletype = {
            options = fakeTilesHelper.getTilesOptions(),
            editable = false
        },
        mode = {
            options = {"Block", "Wall"},
            editable = false
        },
        type = {
            options = {"Bomb", "Drone", "LightningDash", "MegaBomb", "Missile", "RedBooster", "ScrewAttack", "SuperMissile"},
            editable = false
        },
        color = {
            fieldType = "color"
        },
        respawn = {
            options = {"Always", "Until checkpoint", "Never"},
            editable = false
        }
    }
end
BreakBlock.placements = {
    name = "BreakBlock",
    data = {
        width = 8,
        height = 8,
        mode = "Block",
        tiletype = "3",
        flagTiletype = "3",
        flag = "",
        type = "Bomb",
        color = "FFFFFF",
        startRevealed = false,
        directory = "objects/XaphanHelper/BreakBlock",
        respawn = "Always",
        destroyStaticMovers = false
    }
}

BreakBlock.sprite = fakeTilesHelper.getEntitySpriteFunction("tiletype", false, "tilesFg", {1.0, 1.0, 1.0, 0.7})

return BreakBlock

File: Ahorn/entities/BreakBlock.jl
module XaphanHelperBreakBlock

using ..Ahorn, Maple

@mapdef Entity "XaphanHelper/BreakBlock" BreakBlock(x::Integer, y::Integer, width::Integer=8, height::Integer=8, mode::String="Block", tiletype::String="3", type::String="Bomb", color::String="FFFFFF", startRevealed::Bool=false, directory::String="objects/XaphanHelper/BreakBlock")

const placements = Ahorn.PlacementDict(
    "Break Block (Xaphan Helper)" => Ahorn.EntityPlacement(
        BreakBlock,
        "rectangle",
        Dict{String, Any}(),
        Ahorn.tileEntityFinalizer
    ),
)

Ahorn.editingOptions(entity::BreakBlock) = Dict{String, Any}(
    "tiletype" => Ahorn.tiletypeEditingOptions(),
    "mode" => String["Block", "Wall"],
    "type" => String["Bomb", "Drone", "LightningDash", "MegaBomb", "RedBooster", "ScrewAttack"]
)

Ahorn.minimumSize(entity::BreakBlock) = 8, 8
Ahorn.resizable(entity::BreakBlock) = true, true

Ahorn.selection(entity::BreakBlock) = Ahorn.getEntityRectangle(entity)

Ahorn.renderAbs(ctx::Ahorn.Cairo.CairoContext, entity::BreakBlock, room::Maple.Room) = Ahorn.drawTileEntity(ctx, room, entity, alpha=0.7)

end